engine/src/main/java/com/google/android/fhir/MoreDates.kt

```diff
@@ -17,8 +17,13 @@
 package com.google.android.fhir
 
 import java.time.LocalDate
+import java.time.format.DateTimeFormatter
 import java.util.Date
 
 @Suppress("DEPRECATION") // java.util.Date API used by HAPI
 internal val Date.epochDay
   get() = LocalDate.of(year + 1900, month + 1, date).toEpochDay()
+
+/** Validates if given string matches ISO_DATE format i.e. yyyy-MM-dd or yyyy-MM-dd+00:00 */
```

Comment:
nit: this is fine. but i think we normally prefer more explicit comment.

e.g.

```suggestion
/** Returns true if given string matches ISO_DATE format i.e. "yyyy-MM-dd" or "yyyy-MM-dd+00:00", false otherwise. */
```

I also don't necessarily think this should be an extension function to String. I think extension functions should in general be related to the class it's extending. isValidDateOnly isn't a generic string utility. so i think it's better off if we left this just as a general utility.